### ISSUE TITLE
Next.js with SWC causes failure on observeQuery subscriptions

### ISSUE THREAD

**Issue by dpilch** [aws-amplify/amplify-js#10595]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Next.js

### Amplify APIs

DataStore

### Amplify Categories

api

### Environment information

<details>

```
# Put output below this line
  System:
    OS: macOS 13.0
    CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
    Memory: 851.82 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.18.0 - ~/.nvm/versions/node/v14.18.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v14.18.0/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v14.18.0/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Firefox: 102.4.0
    Safari: 16.1
  npmPackages:
    @ampproject/toolbox-optimizer:  undefined ()
    @babel/core:  undefined ()
    @babel/runtime:  7.15.4 
    @edge-runtime/primitives:  2.0.0 
    @hapi/accept:  undefined ()
    @napi-rs/triples:  undefined ()
    @next/react-dev-overlay:  undefined ()
    @segment/ajv-human-errors:  undefined ()
    @types/node: 18.11.8 => 18.11.8 
    @types/react: ^18.0.24 => 18.0.24 
    @types/react-dom: 18.0.8 => 18.0.8 
    @vercel/nft:  undefined ()
    acorn:  undefined ()
    amphtml-validator:  undefined ()
    anser:  undefined ()
    arg:  undefined ()
    assert:  undefined ()
    async-retry:  undefined ()
    async-sema:  undefined ()
    aws-amplify: latest => 4.3.42 
    babel-packages:  undefined ()
    browserify-zlib:  undefined ()
    browserslist:  undefined ()
    buffer:  undefined ()
    bytes:  undefined ()
    chalk:  undefined ()
    ci-info:  undefined ()
    cli-select:  undefined ()
    comment-json:  undefined ()
    compression:  undefined ()
    conf:  undefined ()
    constants-browserify:  undefined ()
    content-disposition:  undefined ()
    content-type:  undefined ()
    cookie:  undefined ()
    cross-spawn:  undefined ()
    crypto-browserify:  undefined ()
    css.escape:  undefined ()
    cssnano-simple:  undefined ()
    data-uri-to-buffer:  undefined ()
    debug:  undefined ()
    devalue:  undefined ()
    domain-browser:  undefined ()
    edge-runtime:  undefined ()
    events:  undefined ()
    find-cache-dir:  undefined ()
    find-up:  undefined ()
    fresh:  undefined ()
    get-orientation:  undefined ()
    glob:  undefined ()
    gzip-size:  undefined ()
    http-proxy:  undefined ()
    https-browserify:  undefined ()
    icss-utils:  undefined ()
    ignore-loader:  undefined ()
    image-size:  undefined ()
    is-animated:  undefined ()
    is-docker:  undefined ()
    is-wsl:  undefined ()
    jest-worker:  undefined ()
    json5:  undefined ()
    jsonwebtoken:  undefined ()
    loader-utils:  undefined ()
    lodash.curry:  undefined ()
    lru-cache:  undefined ()
    micromatch:  undefined ()
    mini-css-extract-plugin:  undefined ()
    nanoid:  undefined ()
    native-url:  undefined ()
    neo-async:  undefined ()
    next: 13.0.1 => 13.0.1 
    node-fetch:  undefined ()
    node-html-parser:  undefined ()
    ora:  undefined ()
    os-browserify:  undefined ()
    p-limit:  undefined ()
    path-browserify:  undefined ()
    platform:  undefined ()
    postcss-flexbugs-fixes:  undefined ()
    postcss-modules-extract-imports:  undefined ()
    postcss-modules-local-by-default:  undefined ()
    postcss-modules-scope:  undefined ()
    postcss-modules-values:  undefined ()
    postcss-preset-env:  undefined ()
    postcss-safe-parser:  undefined ()
    postcss-scss:  undefined ()
    postcss-value-parser:  undefined ()
    process:  undefined ()
    punycode:  undefined ()
    querystring-es3:  undefined ()
    raw-body:  undefined ()
    react: 18.2.0 => 18.2.0 (18.3.0-next-28a574ea8-20221027)
    react-dom: 18.2.0 => 18.2.0 (18.3.0-next-28a574ea8-20221027)
    react-is:  18.2.0 
    react-refresh:  0.12.0 
    react-server-dom-webpack:  undefined ()
    regenerator-runtime:  0.13.4 
    sass-loader:  undefined ()
    scheduler:  undefined ()
    schema-utils:  undefined ()
    semver:  undefined ()
    send:  undefined ()
    setimmediate:  undefined ()
    shell-quote:  undefined ()
    source-map:  undefined ()
    stacktrace-parser:  undefined ()
    stream-browserify:  undefined ()
    stream-http:  undefined ()
    string-hash:  undefined ()
    string_decoder:  undefined ()
    strip-ansi:  undefined ()
    tar:  undefined ()
    terser:  undefined ()
    text-table:  undefined ()
    timers-browserify:  undefined ()
    tty-browserify:  undefined ()
    typescript: 4.8.4 => 4.8.4 
    ua-parser-js:  undefined ()
    undici:  undefined ()
    unistore:  undefined ()
    util:  undefined ()
    vm-browserify:  undefined ()
    watchpack:  undefined ()
    web-vitals:  undefined ()
    webpack:  undefined ()
    webpack-sources:  undefined ()
    ws:  undefined ()
  npmGlobalPackages:
    @aws-amplify/analytics: 5.2.27
    @aws-amplify/api-graphql: 2.3.24
    @aws-amplify/api-rest: 2.0.60
    @aws-amplify/api: 4.0.60
    @aws-amplify/auth: 4.6.13
    @aws-amplify/cache: 4.0.62
    @aws-amplify/cli-internal: 9.2.1
    @aws-amplify/cli: 10.4.0
    @aws-amplify/core: 4.7.11
    @aws-amplify/datastore-storage-adapter: 1.4.4
    @aws-amplify/datastore: 3.14.0
    @aws-amplify/geo: 1.3.23
    @aws-amplify/interactions: 4.1.8
    @aws-amplify/predictions: 4.0.60
    @aws-amplify/pubsub: 4.5.10
    @aws-amplify/pushnotification: 4.3.39
    @aws-amplify/storage: 4.5.13
    @aws-amplify/xr: 3.0.60
    amazon-cognito-identity-js: 5.2.12
    amplify: 0.0.11
    aws-amplify-react-native: 6.0.8
    aws-amplify: 4.3.39
    eslint: 8.24.0
    lerna: 5.5.2
    npm: 8.19.2
    verdaccio: 5.15.3
    yarn: 1.22.19

```

</details>

### Describe the bug

When using Next.js with SWC subscription messages from `observeQuery` result in failure. i.e. when `swcMinify: true` in `next.config.js`. This may happen without Next.js and only SWC, but this has not been tested.

### Expected behavior

Should function identically to when `swcMinify: false`.

### Reproduction steps

1. Create a Next.js
2. Enable DataStore 
3. Add observerQuery subscription.
4. yarn build
5. yarn start (only happens in production build when SWC minifier is used)

### Code Snippet

```javascript
// Put your code below this line.
const sub = DataStore.observeQuery(Todo).subscribe()
```

### Log output

<details>

```
// Put your logs below this line
[WARN] 42:11.65 DataStore - Sync error TypeError: Cannot read properties of undefined (reading 'length')
Uncaught TypeError: Cannot read properties of undefined (reading 'message')

```

</details>

### Additional information and screenshots

As a workaround you can disable the SWC minifier. Set `swcMinify` to `false` in `next.config.js`. https://nextjs.org/docs/messages/swc-minify-enabled

**Comment by dpilch** [comment]
Please comment if you are facing other issues with SWC.

**Comment by svidgen** [comment]
We have another report here: #10732. And this is looking like it will be the new default experience starting with next 13.

**Comment by dpilch** [comment]
A fix was released in [v5.0.7](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify@5.0.7). `swcMinify` can be set to `true` after upgrading.
